From akpm@osdl.org Sat Oct 29 00:07:43 2005
Message-Id: <200510290706.j9T76vbf009061@shell0.pdx.osdl.net>
Subject: USB: usbdevfs_ioctl 32bit fix
To: greg@kroah.com
Cc: akpm@osdl.org, zaitcev@redhat.com
From: akpm@osdl.org
Date: Sat, 29 Oct 2005 00:06:26 -0700


From: Andrew Morton <akpm@osdl.org>

drivers/usb/core/devio.c: In function `proc_ioctl_compat':
drivers/usb/core/devio.c:1401: warning: passing arg 1 of `compat_ptr' makes integer from pointer without a cast

NFI if this is correct...

Cc: Pete Zaitcev <zaitcev@redhat.com>
Signed-off-by: Andrew Morton <akpm@osdl.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


 drivers/usb/core/devio.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/core/devio.c	2005-11-02 09:25:03.000000000 -0800
+++ gregkh-2.6/drivers/usb/core/devio.c	2005-11-02 12:02:56.000000000 -0800
@@ -1392,7 +1392,7 @@
 }
 
 #ifdef CONFIG_COMPAT
-static int proc_ioctl_compat(struct dev_state *ps, void __user *arg)
+static int proc_ioctl_compat(struct dev_state *ps, compat_uptr_t arg)
 {
 	struct usbdevfs_ioctl32 __user *uioc;
 	struct usbdevfs_ioctl ctrl;
@@ -1511,7 +1511,7 @@
 
 	case USBDEVFS_IOCTL32:
 		snoop(&dev->dev, "%s: IOCTL\n", __FUNCTION__);
-		ret = proc_ioctl_compat(ps, p);
+		ret = proc_ioctl_compat(ps, (compat_uptr_t)(long)p);
 		break;
 #endif
 
